src/index.ts

```diff
@@ -1141,21 +1065,21 @@ abstract class PinejsClientCoreTemplate<
 						return `@${option}=${escapeValue(value)}`;
 					}).join('&');
 			} else {
-				paramsObj.url = compiledUrl;
+				params.url = compiledUrl;
 			}
-			const result = this.request(paramsObj);
+			const result = await this.request(params);
 			if (transformFn != null) {
-				return result.then(transformFn) as PromiseResult;
+				return transformFn(result);
 			}
 			return result;
 		};
 	}
 
-	public compile(params: PinejsClientCoreFactory.Params): string {
+	public compile(params: Params): string {
 		if (isString(params)) {
-			deprecated.compileStringParams();
-			return params;
-		} else if (params.url != null) {
+			throw new Error('Params must be an object not a string');
```

Comment:
IMO this is sufficient, but I leave it up to you
```suggestion
			throw new Error('"Params" must be an object');
```